modules/lang-painless/src/test/java/org/elasticsearch/painless/ElvisTests.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless;
+
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+
+/**
+ * Tests for the Elvis operator ({@code ?:}).
+ */
+public class ElvisTests extends ScriptTestCase {
+    public void testBasics() {
```

Comment:
The first one is valid because `?:` unboxes the lhs if it has to. I did this because it felt like the right way to handle a chain like `int i = doc.foo?.bar?.i ?: 0` because `?.` boxes the result so it can be nullable.

The second one is a ClassCastException but that is weird to me because in Java that works....